browser/net/decentralized_dns_network_delegate_helper.cc

```diff
@@ -56,9 +57,56 @@ int OnBeforeURLRequest_DecentralizedDnsPreRedirectWork(
     return net::ERR_IO_PENDING;
   }
 
+  if (IsENSTLD(ctx->request_url) &&
+      IsENSResolveMethodEthereum(g_browser_process->local_state())) {
+    auto* service =
+        brave_wallet::BraveWalletServiceFactory::GetForContext(
+            ctx->browser_context);
+    if (!service) {
+      return net::OK;
+    }
+
+    service->rpc_controller()->EnsProxyReaderResolveAddress(
+        kEnsRegistryContractAddress, ctx->request_url.host(),
+        std::vector<std::string>(std::begin(kRecordKeys),
+                                 std::end(kRecordKeys)),
+        base::BindOnce(&OnBeforeURLRequest_EnsRedirectWork, next_callback,
+                       ctx));
+
+    return net::ERR_IO_PENDING;
+  }
+
   return net::OK;
 }
 
+void OnBeforeURLRequest_EnsRedirectWork(
+    const brave::ResponseCallback& next_callback,
+    std::shared_ptr<brave::BraveRequestInfo> ctx,
+    bool success,
+    const std::string& result) {
+  if (!success) {
+    if (!next_callback.is_null())
+      next_callback.Run();
+    return;
+  }
+  size_t offset = 2 /* len of "0x" */ + 64 /* len of offset to array */;
+  std::string contenthash;
+  if (offset > result.size() ||
+      !brave_wallet::DecodeString(offset, result, &contenthash)) {
```

Comment:
not an issue of this PR, just noticed that it is not really clear what is decoded and how. Maybe we could pick a better name for this